Neos.Media/Classes/Domain/Model/AssetCollection.php

```diff
@@ -177,4 +194,60 @@ public function removeTag(Tag $tag)
         }
         return false;
     }
+
+    public function getParent(): ?self
+    {
+        return $this->parent;
+    }
+
+    public function setParent(?self $parent): void
+    {
+        $this->parent = $parent;
+
+        if ($parent) {
+            // Throw an error if a circular dependency has been detected
+            $parents = [$parent];
+            while ($parent !== null) {
+                $parent = $parent->getParent();
+                if (in_array($parent, $parents, true)) {
+                    throw new \InvalidArgumentException('Circular reference detected', 1680328041);
+                }
+            }
+        }
+    }
+
+    public function addChild(self $child): void
```

Comment:
Don't we have to check for circular references here as well?

Reply:
Fixed thx!
I added additional tests and then fixed the bugs.